docs/machine-learning/tutorials/image-classification-api-transfer-learning.md

```diff
@@ -0,0 +1,456 @@
+
+The images are stored in two sub-directories. Before loading the data, it needs to be format into a list if `ImageInput` objects. To do so, create the `LoadImagesFromDirectory` method below the `Main` method.
+
+```csharp
+public static IEnumerable<ModelInput> LoadImagesFromDirectory(string folder, bool useFolderNameAsLabel = true)
+{
+
+}
+```
+
+1. Inside the `LoadImagesDirectory` add the following code to get all of the file paths from the sub-directories.
+
+    ```csharp
+    var files = Directory.GetFiles(folder, "*",
+        searchOption: SearchOption.AllDirectories);
+    ```
+
+1. Then, iterate through each of the files using a `foreach` statement
+
+    ```csharp
+    foreach (var file in files)
+    {
+
+    }
+    ```
+
+1. Inside the `foreach` statement, check that the file extensions are supported. The Image Classification API supports JPEG and PNG formats.
+
+    ```csharp
+    if ((Path.GetExtension(file) != ".jpg") && (Path.GetExtension(file) != ".png"))
+        continue;
+    ```
+
+1. Then, get the label for the file. If the `useFolderNameAsLabel` parameter is set to `true`, then the parent directory where the file is saved is used as the label. Otherwise, it expects the label to be a prefix of the file name or the file name itself.
+
+    ```csharp
+    var label = Path.GetFileName(file);
+    if (useFolderNameAsLabel)
+        label = Directory.GetParent(file).Name;
+    else
+    {
+        for (int index = 0; index < label.Length; index++)
+        {
+            if (!char.IsLetter(label[index]))
+            {
+                label = label.Substring(0, index);
+                break;
+            }
+        }
+    }
+    ```
+
+1. Finally, create a new instance of `ModelInput`
+
+    ```csharp
+    yield return new ModelInput()
+    {
+        ImagePath = file,
+        Label = label
+    };
+    ```
+
+### Prepare the data
+
+1. Back in the `Main` method, use the `LoadFromDirectory` utility method to get the list of images used for training.
+
+    ```csharp
+    IEnumerable<ModelInput> images = LoadImagesFromDirectory(folder: assetsRelativePath, useFolderNameAsLabel: true);
+    ```
+
+1. Then, load the images into an [`IDataView`](xref:Microsoft.ML.IDataView) using the [`LoadFromEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.LoadFromEnumerable*) method,
+
+    ```csharp
+    IDataView imageData = mlContext.Data.LoadFromEnumerable(images);
+    ```
+
+1. The way in which the data is loaded is in the order it was read from the directories. To balance the data, shuffle it using the [`ShuffleRows`](xref:Microsoft.ML.DataOperationsCatalog.ShuffleRows*) method.
+
+    ```csharp
+    IDataView shuffledData = mlContext.Data.ShuffleRows(imageData);
+    ```
+
+1. It's good practice to split the data into a trianing and validation set. The model is trained on the training set of data and how well it makes predictions on unseen data is measured by the performance against the validation set. Split your data 
```

Comment:
I would say this in stronger terms: it's essential to have a validation set, and you can split the training data into a training and validation set, or you can get a validation set from somewhere else

Reply:
#resolved